### ISSUE TITLE
Remove Date as a Required Field for Descriptive Metadata

### ISSUE THREAD

**Issue by joncameron** [avalonmediasystem/avalon#6021]
# Description
Avalon has required Title and Date as fields that must exist for a record to be validated. This is a bad fit for certain types of metadata and is an artificial requirement that can be removed. Items may only have a creation date, for example, but no publication date as they were never published—but this specific value is the required field, leading people to put in inaccurate data to conform to expectations.

Shouldn't impact facets since null values don't impact index efforts.

Another issue to create: should Creation Date also be used for "Date" indexing as presented in the Blacklight facet search?

# Done Looks Like
- [x] Publication Date is no longer a required field
- [x] Validation for items does not required a Publication Date value
- [x] Documentation updated

**Comment by joncameron** [comment]
A nice idea is adding a date range widget to the facets available in our Blacklight config.

**Comment by joncameron** [comment]
Updated documentation:
- https://samvera.atlassian.net/wiki/spaces/AVALON/pages/1957954564/Metadata+Fields
- https://samvera.atlassian.net/wiki/spaces/AVALON/pages/1957953586/Batch+Ingest+Package+Format
Once #6035 is merged this is ready to close.
